<?xml version="1.0"?>
<launch>

  <arg name="debug" default="false"/>
  <arg name="valgrind" default="false"/>
  <arg name="stack" default="rrbot_stack"/>
  <arg name="spawn" default="true"/>

  <include file="$(find pal_kinematic_simulator)/launch/kinematic_simulator_ros_control.launch">
    <arg name="debug" value="$(arg debug)"/>
    <arg name="valgrind" value="$(arg valgrind)"/>
    <arg name="parameters" value="--cycle_time 0.001"/> 
  </include>

  <!-- Configure wbc -->
  <rosparam command="load"
    file="$(find pal_wbc_tutorial)/config/$(arg stack).yaml" />

 <!-- Configure local joint control -->
  <rosparam command="load"
    file="$(find pal_wbc_tutorial)/config/rrbot_position_joints.yaml" 
    ns="whole_body_kinematic_controller"/>

  <!-- Spawn controller -->
  <group if="$(arg spawn)">
    <node name="wbc_controllers_spawner" pkg="controller_manager" type="spawner" output="screen" args="whole_body_kinematic_controller" />
  </group>

  <!-- Running Joint State Controller -->
  <include file="$(find joint_state_controller)/joint_state_controller.launch"/>

</launch>

